storage-local/src/main/java/io/kestra/storage/local/LocalStorage.java

```diff
@@ -73,33 +82,80 @@ public Long size(String tenantId, URI uri) throws IOException {
 
     @Override
     public Long lastModifiedTime(String tenantId, URI uri) throws IOException {
-        FileTime lastModifiedTime = Files.getLastModifiedTime(getPath(tenantId, uri));
+        FileTime lastModifiedTime;
+        try {
+            lastModifiedTime = Files.getLastModifiedTime(getPath(tenantId, uri));
+        } catch (NoSuchFileException e) {
+            throw new FileNotFoundException(e.getMessage());
+        }
         return lastModifiedTime.toMillis();
     }
 
     @Override
     public URI put(String tenantId, URI uri, InputStream data) throws IOException {
-        this.createDirectory(tenantId, uri);
+        File file = getPath(tenantId, uri).toFile();
+        File parent = file.getParentFile();
+        if (!parent.exists() && !parent.mkdirs()) {
+            throw new RuntimeException("Cannot create directory: " + parent.getAbsolutePath());
+        }
 
-        try (data; OutputStream outStream = new FileOutputStream(getPath(tenantId, uri).toFile())) {
+        try (data; OutputStream outStream = new FileOutputStream(file)) {
             byte[] buffer = new byte[8 * 1024];
             int bytesRead;
             while ((bytesRead = data.read(buffer)) != -1) {
                 outStream.write(buffer, 0, bytesRead);
             }
         }
-
         return URI.create("kestra://" + uri.getPath());
     }
 
+    public FileAttributes getAttributes(String tenantId, URI uri) throws IOException {
+        BasicFileAttributes basicFileAttributes;
+        Path path = getPath(tenantId, uri);
+        try {
+            basicFileAttributes = Files.readAttributes(path, BasicFileAttributes.class);
+        } catch (NoSuchFileException e) {
+            throw new FileNotFoundException(e.getMessage());
+        }
+        return LocalFileAttributes.builder()
+            .fileName(path.getFileName().toString())
+            .basicFileAttributes(basicFileAttributes)
+            .build();
+    }
+
+    @Override
+    public URI createDirectory(String tenantId, URI uri) {
+        File file = getPath(tenantId, uri).toFile();
+        if (!file.exists() && !file.mkdirs()) {
+            throw new RuntimeException("Cannot create directory: " + file.getAbsolutePath());
+        }
+        return URI.create("kestra://" + (uri != null ? uri.getPath() : "/"));
```

Comment:
I don't thinhk we should allow to create a null URI

Reply:
this is only the returned value. Creating "/" means base+tenant+/. But it should not happen as there should always be a prefix so I'll just throw if empty.